datajoint/expression.py

```diff
@@ -441,17 +441,11 @@ def tail(self, limit=25, **fetch_kwargs):
 
     def __len__(self):
         """:return: number of elements in the result set e.g. ``len(q1)``."""
-        try:
-            if self._left[0]:
-                return self.connection.query(
-                    'SELECT distinct count(*) FROM {from_}{where}'.format(
-                        from_=self.from_clause(),
-                        where=self.where_clause())).fetchone()[0]
-        except IndexError:
-            pass
         return self.connection.query(
-            'SELECT count(DISTINCT {fields}) FROM {from_}{where}'.format(
-                fields=self.heading.as_sql(self.primary_key, include_aliases=False),
+            'SELECT {select_} FROM {from_}{where}'.format(
+                select_=('DISTINCT count(*)' if any(self._left)
```

Comment:
`DISTINCT count` does not seem like the right syntax.

Reply:
COUNT(*) returns the number of rows in a specified table, and it preserves duplicate rows. In order to get a proper count of the rows on a left join `DISTINCT` is needed. With a regular count you would put `DISTINCT` inside the parenthesis but COUNT(*) is special and you need to put it outside.

Reply:
Thanks @dimitri-yatsenko. That is right, `DISTINCT` has no effect here and can be removed.

Reply:
As we discussed, `DISTINCT COUNT(*)` does not distinct entries.

Reply:
I have removed `DISTINCT COUNT(*)` in commit 15146b1